templates/kotlin/src/main/kotlin/io/appwrite/models/Payload.kt.twig

```diff
@@ -0,0 +1,71 @@
+package io.appwrite.models
+
+import io.appwrite.extensions.gson
+import java.io.File
+import java.net.URLConnection
+import java.nio.file.Files
+import java.nio.file.Paths
+
+class Payload private constructor() {
+
+    lateinit var path: String
+    lateinit var filename: String
+    lateinit var sourceType: String
+    lateinit var data: Any
+    var mimeType: String? = null
```

Comment:
Do we need `mimetype` and `sourceType`?

Reply:
Its used in Client.kt, so I would say yes. Looking at Payload methods, mimeType is always set internally. So if we ever need to improve it, we can as it wont be user-facing.

sourceType is used to do memory-optimal chunk upload.